File: python/day_1/sonar.py
# pylint: disable=missing-module-docstring

def measurement_increases(inputs):
  counter = 0
  for index, input in enumerate(inputs):
    if index != 0 and input > inputs[index-1]: counter += 1

  return counter

def sliding_window_increases(inputs):
  sums = []
  for index, input in enumerate(inputs[:-2]):
    sums.append(input + inputs[index+1] + inputs[index+2])
  print(sums)
  return measurement_increases(sums)

File: go/day_1/sonar.go
package main

import "fmt"

func main() {
	inputs := []int{2, 3, 5, 7, 11, 13}
	fmt.Println(MeasurementIncreases(inputs))
}

func MeasurementIncreases(inputs []int) int {
	var counter int

	for i := 1; i < len(inputs); i++ {
		if inputs[i] > inputs[i-1] {
			counter++
		}
	}

	return counter
}

func SlidingWindowIncreases(inputs []int) int {
	var sums []int

	for i := 0; i < len(inputs)-2; i++ {
		sums = append(sums, inputs[i]+inputs[i+1]+inputs[i+2])
	}

	return MeasurementIncreases(sums)
}
